cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,128 @@
+            self.db_event = ''
+            self.emit_cmd = self._rename_vm()
+        elif self.action == 'storage':
+            self.tl_event = 'VmReconfiguredEvent'
+            self.tl_category = 'Storage'
+            self.db_event = ''
+            self.emit_cmd = self._rename_vm()
         else:
-            responses = appliance.rest_api.collections.vms.action.add_lifecycle_event(
-                rest_vm, **event)
-        assert appliance.rest_api.response.status_code == 200
-        for response in responses:
-            assert response["success"] is True, "Could not add event"
-
-        # DB check
-        lifecycle_events = appliance.db.client["lifecycle_events"]
-        events_list = list(appliance.db.client.session.query(lifecycle_events).filter(
-            lifecycle_events.message == event["message"],
-            lifecycle_events.status == event["status"],
-            lifecycle_events.event == event["event"],
-        ))
-        assert events_list, "Could not find the lifecycle event in the database"
-
-
-@pytest.mark.manual
-def test_policy_events():
-    pass
-
-
-@pytest.mark.manual
-def test_timelines_ui():
-    pass
+            assert False, logger.warn('VMEvent did not get the proper parameter.')
+
+    def _setup_vm(self):
+        logger.info("Will set up the VM on the provider")
+        self.vm.create_on_provider()
+        return self.vm.provider.mgmt.does_vm_exist(self.vm.name)
+
+    def _power_on(self):
+        logger.info('{} will start.'.format({self.vm.name}))
+        return self.vm.provider.mgmt.start_vm(self.vm.name)
+
+    def _power_off(self):
+        logger.info('{} will stop.'.format({self.vm.name}))
+        return self.vm.provider.mgmt.stop_vm(self.vm.name)
+
+    def _suspend(self):
+        logger.warn("Will Suspend {}!".format(self.vm.name))
+        return self.vm.provider.mgmt.suspend_vm(self.vm.name)
+
+    def _rename_vm(self):
+        logger.warn("Will delete {}!".format(self.vm.name))
+        new_name = self.vm.provider.mgmt.rename_vm(self.vm.name, self.vm.name + '-renamed')
+        if self.vm.name != new_name:
+            self.vm.name = new_name
+        return self.vm.name
+
+    def _add_disk(self):
+        pass
+
+    def _check_timelines(self, target):
+        timelines_view = navigate_to(target, 'Timelines')
+        if self.tl_category not in timelines_view.filter.event_category.all_selected_options:
+            timelines_view.filter.event_category.select_by_visible_text(self.tl_category)
+        timelines_view.filter.time_position.select_by_visible_text('centered')
+        timelines_view.filter.apply.click()
+        found_events = []
+        logger.info(timelines_view.chart.get_events())
+        for evt in timelines_view.chart.get_events():
+            if not hasattr(evt, 'source_vm'):
+                logger.warn("event {} does't have source_vm field.Probably issue".format(evt))
+                continue
+            elif evt.event_type == self.tl_event:
```

Comment:
REQUIRED: this is true only when target is vm. However, you also check timelines for hosts/cluster/etc. So, you have to check that event is present for certain vm.

Reply:
I added this check because it was not working on for event "VM created" because the source VM is the host.